### PR TITLE
More detailed instruction for building from source

### PR DESCRIPTION
**Pull request by spamwax**
Building on the original .md files for building from git/source of Gogs, I have added some step by step instruction on how to properly setup Gogs.

### CODE REVIEW

File: en-US/advanced/configuration_for_source_bulids.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
+---
+name: Detailed configuration for source builds
+sort: 7
+---
+
+Following the instruction of building [from source](/docs/en-US/installation/install_from_source.md), you should now have a local user `git` and a working setup of `go` and [gigs](http://gogs.io/).
+We are gong to setup 'gigs' along with **nginx**. This way, we can take adavantage of nginx. This is useful for servers that already have an nginx running.
```
Comment (Unknwon) [inline en-US/advanced/configuration_for_source_bulids.md]: 1. typo? `gigs`
2. For my OCD, would recommend to use `Nginx`(capital **N**) in all places. :smile:

File: en-US/advanced/configuration_for_source_bulids.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+for example:
+
+```bash
+vi $GOPATH/src/github.com/gogits/gogs/custom/conf/app.ini
+```
+
+## Setting up server
+Starting from top, change the following keys in different sections:
+### main section of app.ini
```
Comment (Unknwon) [inline en-US/advanced/configuration_for_source_bulids.md]: Missing an empty line before this.

File: en-US/advanced/configuration_for_source_bulids.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+### `[repository]` section of app.ini
+
+```bash
+ROOT=/home/git/gogs-repositories
+```
+
+### `[server]` section of app.ini
+```ini
```
Comment (Unknwon) [inline en-US/advanced/configuration_for_source_bulids.md]: Need en empty line after title, so do some other places.

**Comment by Unknwon** [comment]
Never mind the comments I've made, it looks great! I'll do the rest of syntax fix thing.

**Comment by Unknwon** [comment]
You're able to see it here now: http://gogs.io/docs/advanced/configuration_for_source_bulids.html

**Comment by spamwax** [comment]
@Unknwon  Thanks for taking care of typos and links.
I found an issue with link at the bottom of the http://gogs.io/docs/installation/install_from_source.html
where I am linking to the `/docs/advanced/configuration_for_source_builds.md`
Clicking on it takes me to a blank page!

**Comment by Unknwon** [comment]
Fixed, there was a typo in filename, too :smile:

**Pull request merged**